packages/medusa/src/utils/product-category/index.ts

```diff
@@ -0,0 +1,36 @@
+import { FindOptionsWhere } from "typeorm"
+import { ProductCategory } from "../../models"
+import { isDefined } from "medusa-core-utils"
+
+export const categoryMatchesScope = (
+  category: ProductCategory,
+  query: FindOptionsWhere<ProductCategory>
+): boolean => {
+  if (isDefined(query.is_active) && isDefined(query.is_internal)) {
+    return (
+      category.is_internal === query.is_internal &&
+      category.is_active === query.is_active
+    )
+  } else if (isDefined(query.is_active)) {
+    return category.is_active === query.is_active
+  } else if (isDefined(query.is_internal)) {
+    return category.is_internal === query.is_internal
+  } else {
+    return true
+  }
```

Comment:
suggestion:
```ts
return Object.keys(query ?? {}).every(key => category[key] === query[key])
```

Reply:
Correct me if I'm wrong, but this wouldn't handle the AND case in line (9-14) and OR cases in lines 14-16 and 16-18.

Reply:
Oh no, you might be right. If this works, its delish!

Reply:
Nice, that worked! Thanks!

Reply:
Eventually you can add some unit test on that util